docs/install/debug.md

```diff
@@ -369,3 +369,214 @@ Notes:
+
+There is no one recipe for using Intel VTune, as different projects have different needs. Depending on the
+issue to analyze, one might need to configure the experiment in a specific way. However, here are some general
+instructions and tips to get started with using Intel VTune with NEURON.
+
+
+###### Using Caliper VTune Service
+
+If we have installed NEURON with the standard build configuration (e.g. `-DCMAKE_BUILD_TYPE=RelWithDebInfo`) and run
+Vtune with a standard analysis like:
+
+```bash
+vtune -collect uarch-exploration --no-summary -result-dir=nrn_vtune_result x86_64/special -python model.py
+```
+
+and then visualize the results using:
+
+```bash
+vtune-gui nrn_vtune_result &
+```
+
+In the `Bottom-up` analysis, we should see output like this:
+
+![](images/nrn_vtune_uarch_default_bottom_up.png)
+
+This is the expected output, but there are some issues or inconveniences with using the profile in this way:
+
+- The profile will include information from both the model building phase and the simulation phase.
+  While one can `Zoom and Filter by Selection`, the UI is not fluid or user-friendly for such selections.
+- As Vtune is sampling-based, we have functions from the entire execution, making it inconvenient to find
+   the context in which they are called. Importantly, as we have seen in the Caliper profile, we would
+   like to have them grouped in NEURON's terminology and according to hierarchy they are called e.g.
+   "psolve", "timestep", "spike-exchange", "state-update", "hh-state", etc.
+
+This is where Caliper can help us! It uses the Instrumentation and Tracing Technology API, i.e., `ittnotify.h`,
+which we typically use to pause/resume profiling but additionally uses the [Task API](https://github.com/LLNL/Caliper/blob/releases/v2.11.0/src/services/vtune/VTuneBindings.cpp)
+to mark all our Caliper instrumentation regions with `__itt_task_begin()` and `__itt_task_end()`.
+Because of this, all our Caliper instrumentations now nicely appear in Intel Vtune in the views like `Summary`, `Bottom-up`, etc.
+
+![](images/nrn_vtune_uarch_caliper_step1_bottom_up.png)
+
+So let's look at how to build NEURON for Vtune Analysis with Caliper and how to profile execution with various analysis.
+
+
+###### Building NEURON for Vtune Analysis with Caliper
+
+The first step is to install Caliper with Intel VTune support. You can install [Caliper](https://github.com/LLNL/Caliper)
+without Spack, but if you are a Spack user, note that the `vtune` variant in Caliper was added in May 2024. Therefore,
+ensure you are using a newer version of Spack and have the `+vtune` variant activated.
+
+Once Caliper is available with Vtune support, we can build NEURON in a regular way:
+
+```bash
+cmake .. \
+        -DNRN_ENABLE_PROFILING=ON \
+        -DNRN_PROFILER=caliper \
+        -DCMAKE_PREFIX_PATH=/path/to/caliper/install/prefix \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+        -DCMAKE_CXX_FLAGS="-fno-omit-frame-pointer" \
```

Comment:
I'm wondering if we should create build types for this type of situation:
```
cmake -B build-profile -DCMAKE_BUILD_TYPE=VTune .
```

Reply:
Specifically for `-fno-omit-frame-pointer` or to wrap all configurations?

 The former is often needed for all/many profiling tools. We can consider wrapping up some of this into a build type but at least for now, the CMake options/configuration is not that complicated.